src/jit/codegencommon.cpp

```diff
@@ -686,6 +686,13 @@ regMaskTP Compiler::compHelperCallKillSet(CorInfoHelpFunc helper)
     case CORINFO_HELP_STOP_FOR_GC:
         return RBM_STOP_FOR_GC_TRASH;
 
+#ifdef _TARGET_X86_
+    case CORINFO_HELP_INIT_PINVOKE_FRAME:
+        // On x86, this helper has a custom calling convention that takes EDI as argument
+        // (but doesn't trash it), trashes EAX, and returns ESI.
+        return RBM_PINVOKE_SCRATCH | RBM_PINVOKE_TCB;
```

Comment:
How about defining a constant in target.h for this?

e.g.
#define RBM_INIT_PINVOKE_FRAME_TRASH  (RBM_PINVOKE_SCRATCH | RBM_PINVOKE_TCB)

For targets other than x86 it will be RBM_CALLEE_TRASH.

This would not only serve as documentation but also while porting JIT, we just have to define appropriately in target.h.  Otherwise one would have to remember to update this routine which is very easy to miss.  In fact for amd64 unix we had such an instance.